pkg/inputs/snmp/x/arista/eapi.go

```diff
@@ -0,0 +1,374 @@
+	handle.AddCommand(sv)
+	if err := handle.Call(); err != nil {
+		return nil, err
+	}
+
+	dst := kt.NewJCHF()
+	dst.CustomStr = map[string]string{
+		"config_sanity":     sv.ConfigSanity,
+		"local_intf_status": sv.LocalIntfStatus,
+		"neg_status":        sv.NegStatus,
+		"peer_link_status":  sv.PeerLinkStatus,
+		"state":             sv.State,
+		"domain_id":         sv.DomainId,
+		"local_interface":   sv.LocalInterface,
+		"peer_address":      sv.PeerAddress,
+		"peer_link":         sv.PeerLink,
+	}
+	dst.CustomInt = map[string]int32{}
+	dst.CustomBigInt = map[string]int64{}
+	dst.EventType = kt.KENTIK_EVENT_SNMP_DEV_METRIC
+	dst.Provider = c.conf.Provider
+	dst.DeviceName = c.conf.DeviceName
+	dst.SrcAddr = c.conf.DeviceIP
+	dst.Timestamp = time.Now().Unix()
+	dst.CustomMetrics = map[string]kt.MetricInfo{}
+
+	dst.CustomBigInt["ConfigSanity"] = mlag_strings[sv.ConfigSanity]
+	dst.CustomMetrics["ConfigSanity"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["LocalIntfStatus"] = mlag_strings[sv.LocalIntfStatus]
+	dst.CustomMetrics["LocalIntfStatus"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["NegStatus"] = mlag_strings[sv.NegStatus]
+	dst.CustomMetrics["NegStatus"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["PeerLinkStatus"] = mlag_strings[sv.PeerLinkStatus]
+	dst.CustomMetrics["PeerLinkStatus"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["PortsErrdisabled"] = mlag_bool[sv.PortsErrdisabled]
+	dst.CustomMetrics["PortsErrdisabled"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["State"] = mlag_strings[sv.State]
+	dst.CustomMetrics["State"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["PortsDisabled"] = sv.MlagPorts.Disabled
+	dst.CustomMetrics["PortsDisabled"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["PortsActivePartial"] = sv.MlagPorts.ActivePartial
+	dst.CustomMetrics["PortsActivePartial"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["PortsInactive"] = sv.MlagPorts.Inactive
+	dst.CustomMetrics["PortsInactive"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["PortsConfigured"] = sv.MlagPorts.Configured
+	dst.CustomMetrics["PortsConfigured"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
+
+	dst.CustomBigInt["PortsActiveFull"] = sv.MlagPorts.ActiveFull
+	dst.CustomMetrics["PortsActiveFull"] = kt.MetricInfo{Oid: "computed", Mib: "computed", Profile: "eapi.mlag", Type: "eapi.mlag"}
```

Comment:
same as above; should we change the `kt.MetricInfo` to have `Oid: "eapi"`?

Reply:
Good call.